### PR TITLE
Manage `podAntiAffinity` and number of replicas for Control Plane Ingress Controller

### PR DESCRIPTION
**Pull request by TeddyAndrieux**
**Component**:

'cp ingress'

**Context**: 

See: #3574 

**Summary**:

Add an entry in bootstrap configuration for Control Plane Ingress controller in order to set up `affinity`

NOTE: In this PR we only handle soft and hard `podAntiAffinity`

### CODE REVIEW

File: salt/_pillar/metalk8s.py [hunk 1]
```diff
@@ -163,6 +173,19 @@ def _load_iso_path(config_data):
...
 
+def _load_kubernetes(config_data):
+    """Load Kubernetes information from BootstrapConfiguration"""
+    kubernetes_data = config_data.get("kubernetes", {})
+
+    # CoreDNS affinity default to soft podAntiAffinity on hostname
+    kubernetes_data["coreDNS"].setdefault("affinity", {}).setdefault(
+        "podAntiAffinity", {}
+    ).setdefault("soft", [{"topologyKey": "kubernetes.io/hostname"}])
+    kubernetes_data["coreDNS"].setdefault("replicas", 2)
```
Comment (alexandre-allard) [inline salt/_pillar/metalk8s.py]: And if the `coreDNS` entry does not exist ? :)

Comment (TeddyAndrieux) [reply salt/_pillar/metalk8s.py]: Oh right my bad

File: salt/_pillar/metalk8s.py [hunk 2]
```diff
@@ -163,6 +173,19 @@ def _load_iso_path(config_data):
     return res
 
 
+def _load_kubernetes(config_data):
+    """Load Kubernetes information from BootstrapConfiguration"""
+    kubernetes_data = config_data.get("kubernetes", {})
+
+    # CoreDNS affinity default to soft podAntiAffinity on hostname
+    kubernetes_data.etdefault("coreDNS", {}).setdefault("affinity", {}).setdefault(
```
Comment (alexandre-allard) [inline salt/_pillar/metalk8s.py]:
```suggestion
    kubernetes_data.setdefault("coreDNS", {}).setdefault("affinity", {}).setdefault(
```

Comment (TeddyAndrieux) [reply salt/_pillar/metalk8s.py]: :cry:

**Review by alexandre-allard — APPROVED** [review]

**Pull request merged**